azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -398,10 +403,88 @@ public void createContainer(final int executionId) throws ExecutorManagerExcepti
    */
   @Override
   public void deleteContainer(final int executionId) throws ExecutorManagerException {
-    deletePod(executionId);
-    if (isServiceRequired()) {
-      deleteService(executionId);
+    try { // if pod deletion is not successful, the service deletion can still be handled
+      deletePod(executionId);
+    } finally {
```

Comment:
Include `catch` that will log any exceptions during pod and service deletions.

Reply:
In `deletePod ` we're catching and logging exception, and throwing ExecutorManagerException. I'm not sure if we want to catch the exception and log the error again in in `deleteContainer` level.

Reply:
@sshardool I'm addressing this in https://github.com/azkaban/azkaban/pull/3058. Thanks for the suggestion.